server/src-lib/Hasura/Server/Logging.hs

```diff
@@ -269,8 +268,8 @@ mkHttpErrorLogContext userInfoM reqId req err query mTiming compressTypeM header
            , olResponseSize       = Just $ BL.length $ encode err
            , olRequestReadTime    = Seconds . fst <$> mTiming
            , olQueryExecutionTime = Seconds . snd <$> mTiming
-           , olQuery              = either (const Nothing) Just query
-           , olRawQuery           = either (Just . bsToTxt . BL.toStrict) (const Nothing) query
+           , olQuery              = parsedReq
+           , olRawQuery           = Just $ bsToTxt $ BL.toStrict reqBody
```

Comment:
This changes the behavior of this field slightly, but I assume that's fine and that it being null in some cases is not meaningful?

Reply:
Yeah this has a weird behaviour. So the `olQuery` and `olRawQuery` both are `Maybe`s. And previously this function would take in `Either ByteString Value`. The idea was if the request body is parsed (`Right Value`) then we set the `olQuery` field. If we don't have the parsed request body (`Left ByteString`) then we set the `olRawQuery` field. With these changes as we now always have the request body, we should ideally change `olRawQuery` to have always have the value. (And probably it makes sense to call it `olRequestBody` , but that would be a breaking change)

Confirming with @hgiasac (who worked on this), is this right?

I think we can simplify this

Reply:
Sorry, I completely misunderstood your comment! I have restored the behaviour of the field.